src/boot/efi/linux-aarch64.c

```diff
@@ -0,0 +1,161 @@
+        err = uefi_call_wrapper(BS->InstallConfigurationTable, 2,
+                                &EfiDtbTableGuid, fdt);
+        if (EFI_ERROR(err)) {
+                Print(L"Cannot set fdt in EFI configuration\n");
+                return 0;
+        }
+
+        return fdt;
+}
+
+static void *open_fdt(void) {
+        EFI_STATUS status;
+        void *fdt;
+
+        /* Look for a device tree configuration table entry. */
+        status = LibGetSystemConfigurationTable(&EfiDtbTableGuid,
+                                                (VOID**)&fdt);
+        if (EFI_ERROR(status)) {
+                Print(L"DTB table not found, create new one\n");
+                fdt = create_new_fdt(NULL, 2048);
+                if (!fdt)
+                        return 0;
+        }
+
+        if (fdt_check_header(fdt) != 0) {
+                Print(L"Invalid header detected on UEFI supplied FDT\n");
+                return 0;
+        }
+
+        return fdt;
+}
+
+static int update_chosen(void *fdt, UINTN initrd_addr, UINTN initrd_size) {
+        uint64_t initrd_start, initrd_end;
+        int ret, node;
+
+        node = fdt_subnode_offset(fdt, 0, "chosen");
+        if (node < 0) {
+                node = fdt_add_subnode(fdt, 0, "chosen");
+                if (node < 0) {
+                        /* 'node' is an error code when negative: */
+                        ret = node;
+                        Print(L"Error creating chosen\n");
+                        return ret;
+                }
+        }
+
+        initrd_start = cpu_to_fdt64(initrd_addr);
+        initrd_end = cpu_to_fdt64(initrd_addr + initrd_size);
+
+        ret = fdt_setprop(fdt, node, "linux,initrd-start",
+                          &initrd_start, sizeof initrd_start);
+        if (ret) {
+                Print(L"Cannot create initrd-start property\n");
+                return ret;
+        }
+
+        ret = fdt_setprop(fdt, node, "linux,initrd-end",
+                          &initrd_end, sizeof initrd_end);
+        if (ret) {
+                Print(L"Cannot create initrd-end property\n");
+                return ret;
+        }
+
+        return 0;
+}
+
+#define FDT_EXTRA_SIZE 0x400
+
+/* Update fdt /chosen node with initrd address and size */
+static void update_fdt(UINTN initrd_addr, UINTN initrd_size) {
+        void *fdt;
+
+        fdt = open_fdt();
+        if (fdt == 0)
+                return;
+
+        if (update_chosen(fdt, initrd_addr, initrd_size) == -FDT_ERR_NOSPACE) {
+                /* Copy to new tree and re-try */
+                Print(L"Not enough space, creating a new fdt\n");
+                fdt = create_new_fdt(fdt, fdt_totalsize(fdt) + FDT_EXTRA_SIZE);
+                if (!fdt)
+                        return;
+                update_chosen(fdt, initrd_addr, initrd_size);
+        }
+}
+
+/* linux_addr is the .linux section address */
+/* We don't use cmdline in aarch64 (kernel EFI stub takes it itself from the
+ * EFI LoadOptions) */
```

Comment:
not sure i follow. plesae note the logic implemented in stub.c, which does different things with cmdline depending on secureboot. If you ignore the param here, then this logic won't work

Reply:
Indeed, this breaks measurements, reality and expectations as to whether or not `.cmdline` args are measured and used.